numpy-stubs/__init__.pyi

```diff
@@ -489,5 +490,160 @@ def ones(
     shape: _ShapeLike, dtype: _DtypeLike = ..., order: Optional[str] = ...
 ) -> ndarray: ...
 
+class ufunc:
+    def __call__(
+        self,
+        *args: _ArrayLike,
+        out: Optional[Union[ndarray, Tuple[ndarray, ...]]] = ...,
+        where: Optional[ndarray] = ...,
+        # The list should be a list of tuples of ints, but since we
+        # don't know the signature it would need to be
+        # Tuple[int, ...]. But, since List is invariant something like
+        # e.g. List[Tuple[int, int]] isn't a subtype of
+        # List[Tuple[int, ...]], so we can't type precisely here.
+        axes: List[Any] = ...,
+        axis: int = ...,
+        keepdims: bool = ...,
+        # TODO: make this precise when we can use Literal.
+        casting: str = ...,
+        # TODO: make this precise when we can use Literal.
+        order: Optional[str] = ...,
+        dtype: Optional[_DtypeLike] = ...,
+        subok: bool = ...,
+        signature: Union[str, Tuple[str]] = ...,
+        # In reality this should be a length of list 3 containing an
+        # int, an int, and a callable, but there's no way to express
+        # that.
+        extobj: List[Union[int, Callable]] = ...,
+    ) -> Union[ndarray, generic]: ...
```

Comment:
I think that after a mypy upgrade to resolve this issue

https://github.com/python/typing/issues/253

the union can be removed and replaced with two overloads (one for all scalars and one for array like). On the pinned version of mypy that doesn't work because the signatures have overlapping input types.